.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,198 @@
+name: Build, test and deploy
+
+on:
+  push:
+    branches: [ master ]
```

Comment:
> can we use a variable for main branch; in case it changes ?

Do you have a strategy in mind for accomplishing this? I tried a few things and I can't make this work:

- environment variables seem to be silently ignored at the top level (at least in this context) - my builds never ran
- YAML anchors and aliases aren't supported (ref [here](https://github.community/t/support-for-yaml-anchors/16128))

Reply:
how about we just simplify this to:

```
on: [ push, pull_request ]
```

Reply:
I've left the option of manual triggering in, but this is otherwise resolved. The justification for manual triggering is in the event of an unexpected build failure on master, which is possible given the update and checkout actions contained in the workflow.